src/arrayMethodSort.js

```diff
@@ -1,11 +1,42 @@
+/* eslint-disable no-param-reassign */
 'use strict';
 
 /**
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function (compareFunction) {
+    if (compareFunction === undefined) {
+      compareFunction = (a, b) => {
+        const isStringA = typeof a === 'string';
+        const isStringB = typeof b === 'string';
+
+        if (isStringA && isStringB) {
+          const isUpperA = a[0] === a[0].toUpperCase();
+          const isUpperB = b[0] === b[0].toUpperCase();
+
+          if (isUpperA && !isUpperB) {
+            return -1;
+          }
+
+          if (!isUpperA && isUpperB) {
+            return 1;
+          }
+        }
+
+        return String(a).localeCompare(String(b));
```

Comment:
The default compare function should compare strings based on Unicode code points, not by checking capitalization. The current implementation with capitalization check is not aligned with the default behavior of Array.prototype.sort.